api/secrets.proto

```diff
@@ -0,0 +1,106 @@
+syntax = "proto3";
```

Comment:
For secrets, at least, we would like the option to have different backends (such as vault or whatever someone is already using for secrets management).  Being able to swap that or have third parties implement a pluggable backends would be very useful.